UnityProject/Assets/Scripts/Util/SingletonManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿using UnityEngine;
+
+public class SingletonManager<T> : MonoBehaviour where T : MonoBehaviour
```

Comment:
@ThatDan123 I remember you said something about the pattern new singletons should follow, can you take a look at this?

Reply:
i have to work on this for a while since with the changes needed it requires alot of additions ill open it at a later time

Reply:
you could have ommitted the namespaces changes, at least for classes that weren't new. Anyways, don't be afraid to make smaller PRs and progresively implement your changes the next time!

Reply:
sure thing i mostly just want to implement the singletonManager since there are quite a few managers which require it

Reply:
also as far as I can see the project itself requires a lot of restructure

Reply:
honestly, yeah. Any help on that regard would make us happy.